KerbalVR_Mod/KerbalVR/KerbalVR_InteractionSystem.cs

```diff
@@ -1,121 +1,138 @@
-        //protected KerbalVR.HeadUpDisplay hud;
-
-        // device behaviors and actions
-        protected bool isHandsInitialized = false;
-        protected SteamVR_Action_Pose handActionPose;
-        #endregion
-
-
-        protected void InitializeHandScripts() {
-            // store actions for these devices
-            handActionPose = SteamVR_Input.GetPoseAction("default", "Pose");
-
-            // set up the hand objects
-            var lhGameObject = new GameObject("KVR_HandL");
-            lhGameObject.transform.SetParent(transform, false);
-            DontDestroyOnLoad(lhGameObject);
-            LeftHand = lhGameObject.AddComponent<KerbalVR.Hand>();
-            LeftHand.handPrefab = glovePrefabL;
-            LeftHand.handType = SteamVR_Input_Sources.LeftHand;
-            LeftHand.handActionPose = handActionPose;
-
-            var rhGameObject = new GameObject("KVR_HandR");
-            rhGameObject.transform.SetParent(transform, false);
-            DontDestroyOnLoad(rhGameObject);
-            RightHand = rhGameObject.AddComponent<KerbalVR.Hand>();
-            RightHand.handPrefab = glovePrefabR;
-            RightHand.handType = SteamVR_Input_Sources.RightHand;
-            RightHand.handActionPose = handActionPose;
-
-            RightHand.otherHand = LeftHand.gameObject;
-            LeftHand.otherHand = RightHand.gameObject;
-
-            // can init the skeleton behavior now
-            LeftHand.Initialize();
-            RightHand.Initialize();
-
-            // init the head up display
-            //HeadUpDisplay = new GameObject("KVR_HeadUpDisplay");
-            //DontDestroyOnLoad(HeadUpDisplay);
-            //hud = HeadUpDisplay.AddComponent<KerbalVR.HeadUpDisplay>();
-            //hud.Initialize();
-        }
-
-    } // class InteractionSystem
+		#endregion
+
+		private void OnDestroy()
+		{
+			Utils.LogError("Interaction System being destroyed");
+		}
+
+		#region Properties
+
+		private bool useIVAProfile = false;
+		public bool UseIVAProfile
+		{
+			get => useIVAProfile;
+			set
+			{
+				useIVAProfile = value;
+
+				if (useIVAProfile)
+				{
+					LeftHandIVA.gameObject.SetActive(true);
+					RightHandIVA.gameObject.SetActive(true);
+					LeftHandEVA.gameObject.SetActive(false);
+					RightHandEVA.gameObject.SetActive(false);
+				}
+				else
+				{
+					LeftHandIVA.gameObject.SetActive(false);
+					RightHandIVA.gameObject.SetActive(false);
+					LeftHandEVA.gameObject.SetActive(true);
+					RightHandEVA.gameObject.SetActive(true);
+				}
+
+				LeftHandIVA.ChangeGrab(false);
+				RightHandIVA.ChangeGrab(false);
+				LeftHandEVA.ChangeGrab(false);
+				RightHandEVA.ChangeGrab(false);
+			}
+		}
+
+		public Hand LeftHand { get => UseIVAProfile ? LeftHandIVA : LeftHandEVA; }
+		public Hand RightHand { get => UseIVAProfile ? RightHandIVA : RightHandEVA; }
+
+		private Hand LeftHandIVA;
+		private Hand LeftHandEVA;
+		private Hand RightHandIVA;
+		private Hand RightHandEVA;
```

Comment:
Does it really make sense to have 4 hands, or is it reasonable to just create/destroy/retarget the components that care about the model and colliders?  I worry that some things will break if they don't take into account that the logical hand object is changing - like what it's attached to, etc.

This might still be the best way, just something to think about.

Reply:
I need to dig into this a little more too - is it reasonable to only destroy/recreate the actual hand gameobject (the thing with the renderers etc) and leave the gameobject with the tracking behaviors alive forever? Or are there things in the tracking components that differ per hand model (e.g. for skeleton stuff?)

Reply:
The specific kind of thing I'm thinking about is here: https://github.com/JonnyOThan/Kerbal-VR/blob/3beeca1dd976b8db1be31ec4490c06b68c81ecbf/KerbalVR_Mod/KerbalVR/KerbalVR_UISystem.cs#L287